### PR TITLE
feat(onboarding): mobile onboarding redirect view

### PR DESCRIPTION
**Pull request by PixelDust22**
image
image

### CODE REVIEW

File: static/app/views/onboarding/targetedOnboarding/components/createProjectsFooter.tsx [hunk 1]
```diff
@@ -79,7 +82,30 @@ export default function CreateProjectsFooter({
...
+      if ((navigator as any).userAgentData) {
+        isMobile = (navigator as any).userAgentData.mobile;
+      }
+      if (
+        isMobile &&
+        organization.experiments.TargetedOnboardingMobileRedirectExperiment === 'hide'
+      ) {
+        if (shouldSendEmail) {
+          persistedOnboardingState &&
+            api.requestPromise(
```
Comment (scefali) [inline static/app/views/onboarding/targetedOnboarding/components/createProjectsFooter.tsx:95]: @Neo-Zhixing maybe we should wait until this succeeds before sending the user somewhere else

File: static/app/views/onboarding/targetedOnboarding/components/createProjectsFooter.tsx [hunk 2]
```diff
@@ -79,7 +82,30 @@ export default function CreateProjectsFooter({
         organization,
       });
       clearIndicators();
-      onComplete();
+      let isMobile: boolean = window.innerWidth < 800;
+      if ((navigator as any).userAgentData) {
+        isMobile = (navigator as any).userAgentData.mobile;
```
Comment (scefali) [inline static/app/views/onboarding/targetedOnboarding/components/createProjectsFooter.tsx:87]: @Neo-Zhixing can we make this into a utility function?

File: static/app/views/onboarding/targetedOnboarding/onboarding.tsx [hunk 3]
```diff
@@ -192,7 +203,7 @@ function Onboarding(props: Props) {
               <stepObj.Component
                 active
                 stepIndex={stepIndex}
-                onComplete={() => goNextStep(stepObj)}
+                onComplete={() => stepObj && goNextStep(stepObj)}
```
Comment (scefali) [inline static/app/views/onboarding/targetedOnboarding/onboarding.tsx:206]: @Neo-Zhixing If `stepObj` is falsy, wouldn't `stepObj.Component` just error out anyways?

**Review by scefali — COMMENTED** [review]
@Neo-Zhixing I think we are going to need logic so that all mobile users end up with the targeted onboarding component controlled here: https://github.com/getsentry/sentry/blob/66b5c27a09599aeab7eeb316fe5c1089a1176df9/static/app/views/onboarding/onboardingController.tsx#L18

Otherwise, mobile users may not actually see the version of onboarding we ant.

File: static/app/views/onboarding/targetedOnboarding/onboarding.tsx (hunk 3 above)
Comment (Neo-Zhixing) [reply static/app/views/onboarding/targetedOnboarding/onboarding.tsx:206]: Yeah so this is just there to make TypeScript happy.

**Comment by Neo-Zhixing** [comment]
> @Neo-Zhixing I think we are going to need logic so that all mobile users end up with the targeted onboarding component controlled here:
> 
> https://github.com/getsentry/sentry/blob/66b5c27a09599aeab7eeb316fe5c1089a1176df9/static/app/views/onboarding/onboardingController.tsx#L18
> 
> Otherwise, mobile users may not actually see the version of onboarding we ant.

Do we want to run this experiment concurrently with the MultiPlatform Select experiment or can we make a decision before launch?

**Comment by scefali** [comment]
> > @Neo-Zhixing I think we are going to need logic so that all mobile users end up with the targeted onboarding 
> Do we want to run this experiment concurrently with the MultiPlatform Select experiment or can we make a decision before launch?
I would say no, but we should add the code for the experiment before it's actually running so we can test it (without logging the experiment of course). We can end the current experiment in a couple of days when the mobile one is ready to go.

**Review by scefali — APPROVED** [review]

**Pull request merged**